### PR TITLE
TextEdit update cache.size on ENTER_TREE

### LINKED ISSUE #20296: TextEdit not showing typed text and error on scrolling/typing
**Godot version:**
Tried with:
v3.1.dev.custom_build.b2d7641
v3.1.dev.custom_build.43dc050
v3.1.dev.custom_build.911febb

Every time that I will use Godot I make a build

**OS/device including version:**
Windows 10 Pro
GeForce GTX 1080/PCIe/SSE2

**Issue description:**
Can't use the TextEdit properly.
It was supposed to show the text that I type or the text on _Text_ property

**Steps to reproduce:**
Create a blank project, place a root node (I tried with _Node_ or _Control_), create a TextEdit child node and run the project.
With the TextEdit on focus, scroll down and then "scroll more" to get the first debug errors
Typing after that scroll does trigger the error too.

Error:
> 0:00:04:0295 - Index line=3 out of size (text.size()=3)
> Type:Error Description: Time: 0:00:04:0295 C Error: Index line=3 out of size (text.size()=3) C Source: 
> scene\gui\text_edit.cpp:3731 C Function: TextEdit::times_line_wraps

**Minimal reproduction project:**
[TextEditTest.zip](https://github.com/godotengine/godot/files/2212381/TextEditTest.zip)

[Video of the problem](https://www.youtube.com/watch?v=44m4fBDgN_0)

### PR DESCRIPTION
**Pull request by AlexHolly**
closes https://github.com/godotengine/godot/issues/20296

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @groud, it's partially undoing #19960.

File: scene/gui/control.cpp [hunk 1]
```diff
@@ -1343,11 +1343,13 @@ void Control::_size_changed() {
 
 	data.pos_cache = new_pos_cache;
 	data.size_cache = new_size_cache;
+
```
Comment (akien-mga) [inline scene/gui/control.cpp]: Trailing whitespace.

**Comment by groud** [comment]
I am not sure this is a good solution. There's quite no reason why the notification should be sent to children if the node is not inside the tree. But maybe I'm wrong, I'm not sure.

**Comment by AlexHolly** [comment]
I will try something else. It is strange that it happens only in TextEdit.

**Comment by Zylann** [comment]
Isn't sending notifications from the scene tree to nodes that are not in that tree is breaking thread-safety rules?

**Comment by guilhermefelipecgs** [comment]
If we just add this `set_anchor(MARGIN_BOTTOM, ANCHOR_END);` to TextEdit seems to work.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**